python/pyspark/sql/session.py

```diff
@@ -906,6 +907,12 @@ def dataSource(self) -> "DataSourceRegistration":
 
         return DataSourceRegistration(self)
 
+    @property
+    def profile(self) -> "Profile":
```

Comment:
Need this for connect, too?

Reply:
I planned to make a follow-up PR for Connect. But now I think I'll just add Connect changes to this PR.